### PR TITLE
Feature vsp prop

### PR DESCRIPTION
**Pull request by rachealerhard**
Updates vsp_read to include a read in of all propellers and lift rotors
Adds script for reading in propeller/rotor geometry from OpenVSP

### CODE REVIEW

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    prop.max_thickness_distribution = prop.thickness_to_chord*prop.chord_distribution * units_factor
+
+    prop.Cl_distribution            = np.array(vsp.GetDoubleResults(rid, "CLi" ))
+
+    number_of_radial_stations        = len(prop.chord_distribution)
+    
+    # Extra data from VSP BEM that SUAVE might not use...
+    prop.beta34 		= vsp.GetDoubleResults(rid, "Beta34" )[0]  # pitch at 3/4 radius
+    prop.pre_cone 		= vsp.GetDoubleResults(rid, "Pre_Cone")[0]
+    prop.rake 			= np.array(vsp.GetDoubleResults(rid, "Rake"))
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py:161]: I know most BEMT's can't handle it, but rake and skew would be nice in the future

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    prop.axial 			= np.array(vsp.GetDoubleResults(rid, "Axial"))
+    prop.tangential 		= np.array(vsp.GetDoubleResults(rid, "Tangential"))
+
+    # Set prop rotation
+    prop.rotation = 1
+    
+    # ---------------------------------------------
+    # Rotor Airfoil
+    # ---------------------------------------------
+    if write_airfoil_file:
```
Comment (planes) [inline trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py:172]: Just let it default to SUAVE's DAE-51.

I realize the DAE-51 airfoil and fit isn't perfect. But the upside of assuming the NACA 4412 is minimal in comparison to the downside of having us distribute polars.

File: trunk/SUAVE/Input_Output/OpenVSP/Airfoils/Polars/Clark_y_polar_Re_100000.txt [hunk 3]
```diff
@@ -0,0 +1,125 @@
+
```
[the remark this answers is not in the document]
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/Airfoils/Polars/Clark_y_polar_Re_100000.txt:1]: Removed airfoil folder, so it now defaults to DAE-51 in rotor spin function

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read.py [hunk 4]
```diff
@@ -169,4 +176,39 @@ def vsp_read(tag, units_type='SI'):
...
+			propellers.append(prop)
+			number_of_propeller_engines += 1 
+
+	# Create the rotor network
+	if number_of_lift_rotor_engines>0 and number_of_propeller_engines>0:
+		# Lift + Cruise network
+		net = Lift_Cruise()
+	else:
+		net = Battery_Propeller()
+	if number_of_lift_rotor_engines>0:
```
[the remark this answers is not in the document]
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/vsp_read.py]: Updated to account for battery propeller networks that have all lift rotors assigned as propellers

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    Inputs:
+    1. VSP 10-digit geom ID for prop.
+    2. units_type set to 'SI' (default) or 'Imperial'.
+    3. write_airfoil_file set to True (default) or False
+    4. number_of_radial_stations is the radial discretization used to extract the propeller design from OpenVSP
+
+    Outputs:
+    Writes SUAVE propeller/rotor object, with these geometries, from VSP:
+    	prop.
+    		origin                                  [m] in all three dimensions
```
[the remark this answers is not in the document]
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py:52]: done

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        parm_name = vsp.GetParmName(parm_id[i])
+        parm_names.append(parm_name)
+
+    # Run the vsp Blade Element analysis
+    vsp.SetStringAnalysisInput( "BladeElement" , "PropID" , (prop_id,) )
+    rid = vsp.ExecAnalysis( "BladeElement" )
+    Nc  = len(vsp.GetDoubleResults(rid,"YSection_000"))
+    prop.number_points_around_airfoil = 2*Nc
+    
+    prop.CLi			    = vsp.GetParmVal(parm_id[parm_names.index('CLi')])
```
[the remark this answers is not in the document]
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py]: done

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py (hunk 2 above)
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py:171]: Fixed

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py (hunk 1 above)
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py:160]: Agreed

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read.py [hunk 7]
```diff
@@ -169,4 +176,47 @@ def vsp_read(tag, units_type='SI'):
...
+			net.lift_rotors.append(propellers[list(lift_rotors.keys())[i]])
+		net.number_of_lift_rotor_engines = number_of_lift_rotor_engines	
+		
+		for i in range(number_of_propeller_engines):
+			net.propellers.append(propellers[list(propellers.keys())[i]])
+		net.number_of_propeller_engines = number_of_propeller_engines		
+		
+	else:
+		# Append all rotors as propellers for the battery propeller network
+		net = Battery_Propeller()
```
Comment (mclarke2) [inline trunk/SUAVE/Input_Output/OpenVSP/vsp_read.py:209]: is there a way to pass the type of network the user wants to specify into the vsp_read function? That way we do not assume it is a battery propeller.

**Review by mclarke2 — COMMENTED** [review]
Nice PR! Just one major conceptual change i'd suggest

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_read.py (hunk 7 above)
Comment (rachealerhard) [reply trunk/SUAVE/Input_Output/OpenVSP/vsp_read.py]: Updated to add a check for specified network; uses this convention if no network is specified

**Review by mclarke2 — APPROVED** [review]
Looks great!

**Pull request merged**